schema-registry/common/src/main/java/com/hortonworks/registries/schemaregistry/ConfigEntry.java

```diff
@@ -128,6 +137,44 @@ public String toString() {
                 '}';
     }
 
+    /**
+     * Parses a String to a typed value
+     *
+     * @param <T> type of the resulting value.
+     */
+    public interface Converter<T> {
+        T convert(String str);
```

Comment:
Maybe we should take Object instead of String here as input for convert(). I believe it makes the usage of convert() cleaner in the SchemaRegistryClient. In schema registry client we won't have to typecast Object into String and then call convert() and we won't have to check if the input is an "instance of" of String and then invoke the convert, all these checks and conversions can be pushed into the convert() function itself.

Reply:
Good point, I'm going to change it.

Reply:
I'm going to follow your comment. 
However, I'd like to ask a question. Would it be worth considering typed Property handling, such as Jackson Text dataformats (see: https://github.com/FasterXML/jackson-dataformats-text/tree/master/properties )